### PR TITLE
Maintenance/lint

### PR DESCRIPTION
**Pull request by jcampbell**
Updates our linting configuration to fail azure pipelines when linting is incomplete, and brings repo back to nominal standard.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -46,11 +49,13 @@ stages:
...
 
           - script: |
-              pip install  isort[requirements]==5.4.2 flake8==3.8.3 black==19.10b0 pyupgrade==2.7.2
-              isort . --check-only --skip docs/
-              black --check --exclude docs/ .
-              flake8 great_expectations/core
-              pyupgrade --py3-plus
+              pip install isort[requirements]==5.4.2 flake8==3.8.3 black==19.10b0 pyupgrade==2.7.2
+              EXIT_STATUS=0
+              isort . --check-only --skip docs/ || EXIT_STATUS=$?
```
Comment (anthonyburdi) [inline azure-pipelines.yml:51]: Very cool! Like the use of `||`

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 trigger:
   - develop
   - main
+  - 0.*-develop
+  - feature_dev_wip/partitioners
```
Comment (anthonyburdi) [inline azure-pipelines.yml]: I assume the plan is to remove these feature branches when they are merged. Could these two lines be changed to the following pattern if we prefix all feature branches with `feature`?
```suggestion
  - feature*
```

**Review by anthonyburdi — APPROVED** [review]
Just a suggestion but feel free to merge with or without!

File: azure-pipelines.yml (hunk 2 above)
Comment (jcampbell) [reply azure-pipelines.yml]: Good catch -- these should be removed entirely

Comment (jcampbell) [reply azure-pipelines.yml]: Updated.

**Pull request merged**